### PR TITLE
GCS_MAVLink: check reply channel for space rather than current channel

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
On Sun, 27 Oct 2019, Andrew Tridgell wrote:

> I think we need a peek here. Otherwise if there isn't enough space then we've still popped the entry from the queue and lost it

Fixed.

File: libraries/GCS_MAVLink/GCS_Param.cpp [hunk 1]
```diff
@@ -433,6 +427,11 @@ uint8_t GCS_MAVLINK::send_parameter_async_replies()
 
         _queued_parameter_send_time_ms = AP_HAL::millis();
         async_replies_sent_count++;
+
+        if (!param_replies.pop()) {
+            // internal error...
```
Comment (WickedShell) [inline libraries/GCS_MAVLink/GCS_Param.cpp]: SITL panic?

**Comment by peterbarker** [comment]
On Sun, 27 Oct 2019, WickedShell wrote:

> SITL panic?

Pre-existing code, no intention to rabbit-hole on this one.

And it could *actually* be a generic flow-of-control internal error 
call...

**Comment by rmackay9** [comment]
this has been added to Copter-4.0.0-rc3

**Pull request merged**